tests/rust_usage_test/bin/alloc_check.rs

```diff
@@ -126,8 +126,8 @@ fn main() {
 
             let test4 = m.test4().unwrap();
             assert_eq!(test4.len(), 2);
-            assert_eq!(test4[0].a() as i32 + test4[0].b() as i32 +
-                       test4[1].a() as i32 + test4[1].b() as i32, 100);
+            assert_eq!(i32::from(test4[0].a() + test4[1].a()) +
```

Comment:
@CasperN This appears to change the meaning of the code, because the casts are happening twice, not four times. I don't mind it in general, but it does change what the code is doing.

Reply:
Yep

Reply:
@CasperN was this the result of a lint? The change here makes it possible to overflow, where it couldn't before. This is what I mean when I ask about the difference between semantics vs syntax change.

Reply:
You're right. The lint is about preferring `from` over `as` - in case types change and `as` becomes silently lossy. I wrote it out with additions inside `from(..)` to save on characters but it does introduce risk of overflow, so I wrote it out the verbose way.

https://rust-lang.github.io/rust-clippy/master/#cast_lossless

Reply:
@CasperN Ah, I see. That is a positive change. Can you slightly modify this wrap every call to `a` or `b` in a `from` and call it done? Then it stays lossless, and doesn't change the meaning of the code. Thanks! :-)